src/Swarm/Util.hs

```diff
@@ -185,6 +198,15 @@ isIdentChar c = isAlphaNum c || c == '_' || c == '\''
 replaceLast :: Text -> Text -> Text
 replaceLast r t = T.append (T.dropWhileEnd isIdentChar t) r
 
+-- | Fail with a Text-based message, made out of phrases to be joined
+--   by spaces.
+failT :: MonadFail m => [Text] -> m a
+failT = fail . from @Text . T.unlines
```

Comment:
Should be `T.unwords`?

Reply:
Yes, it should!  See, this is exactly why I didn't want to merge it yet - I wanted you to look at the new code first!